### PR TITLE
added miniseq parsing

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
Adding support for the MiniSeq

### CODE REVIEW

**Comment by JeffreyThiessen** [comment]
Once again, I'm not sure why merging dev into this branch is not collapsing the commit list. A squash and merge would probably be good for this one too.

**Comment by JeffreyThiessen** [comment]
Fixed the git history issues, shouldn't need squash anymore.

File: parsers/miniseq/parser.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            logging.error(("The directory is not accessible, can not parse samples from this directory {}"
+                           "".format(directory), directory))
+            raise exceptions.DirectoryError("The directory is not accessible, "
+                                            "can not parse samples from this directory {}".format(directory), directory)
+
+        sample_sheet_file_name = Parser.SAMPLE_SHEET_FILE_NAME
+        file_list = next(os.walk(directory))[2]  # Gets the list of files in the directory
+        if sample_sheet_file_name not in file_list:
+            logging.error("No sample sheet file in the MiSeq format found")
+            raise exceptions.DirectoryError("The directory {} has no sample sheet file in the MiSeq format"
```
Comment (tom114) [inline parsers/miniseq/parser.py]: Change `MiSeq` here and above to `MiniSeq`

File: parsers/miniseq/sample_parser.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+            pf_list = list(filter(regex.search, data_dir_file_list))
+
+            if not pf_list:
+                # we **still** didn't find anything. It's pretty likely, then that
+                # there aren't any fastq files in the directory that match what
+                # the sample sheet says...
+                raise exceptions.SequenceFileError(
+                    ("The uploader was unable to find an files with a file name that ends with "
+                     ".fastq.gz for the sample in your sample sheet with name {} in the directory {}. "
+                     "This usually happens when the Illumina MiSeq Reporter tool "
```
Comment (tom114) [inline parsers/miniseq/sample_parser.py]: More `MiSeq`

File: parsers/miniseq/sample_parser.py [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+        if len(sample_dict.keys()) != len(line):
+            """
+            if there is one more Data header compared to the length of
+            data values then add an empty string to the end of data values
+            i.e the Description will be empty string
+            assumes the last Data header is going to be the Description
+            this handles the case where the last trailing comma is trimmed
+
+            Shaun said this issue may come up when a user edits the
+            SampleSheet from within the MiSeq software
```
Comment (tom114) [inline parsers/miniseq/sample_parser.py]: More MiSeq

**Pull request merged**